kubernetes_deploy/development/service.yml

```diff
@@ -11,3 +11,20 @@ spec:
     nodePort: 31000
   selector:
     app: laa-legal-adviser-api
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: laa-legal-adviser-api-mq
+spec:
+  selector:
+    app: laa-legal-adviser-api
```

Comment:
🙋‍♂ This needs to be a selector for the pod providing RabbitMQ.

Reply:
On development that pod has the application, rabbitmq and postgress containers and that has a name of laa-legal-adviser-api

Reply:
I see! If they're in the same pod then they can each talk to each other via hostnames (all containers in the same pod share the same network namespace).

However, is that a good model? That would mean each pod would have a separate RabbitMQ instance.

Reply:
>we should make sure there's a maximum replica count with a comment to avoid surprises.

Do you mean changing the following in the deployment.yml replica:
```  # This pod contains the app, database and rabbitmq containers. Keep this count to 1 otherwise you will have multiple rabbitmq instances, alternatively you can create a pod for each of these containers
  replicas: 1```

Reply:
Yeah, that can work; it still creates a surprise when someone scales up through the `kubectl` command locally.

That said, if we use containers, we don't need a `Service` as every container in a pod can talk to the other containers.

Reply:
The app,db,and rabbitmq are in one pod.

The workers are in another pod. So you still need a service for the workers to communicate with the db and rabbitmq containers in another pod.

Are you suggesting moving everything into pod and removing the services?

Reply:
Right, I missed (again) that there are two pods now.

My thinking is: if the database and messaging broker has to be shared by the 
API/admin and the worker, it feels weird to use the API/admin to connect to those components.

So I figure we could replicate the pod configuration on staging/production and add additional pods for the services (move the db/mq to a separate pod) _or_ put everything in the same pod with a comment to not scale up.

Reply:
I like the first option of moving the shared services(db/mq) into a separate pod

Reply:
This pull request already delivers value to the staging configuration. What do you think about splitting the pull request to "staging" and "development" changes? Is it worth it?

Reply:
Or taking it as-is and adding a new pull request on top?

Reply:
Added `kubernetes_deploy/development/shared_services_deployment.yml` depoyment that has the database and rabbitmq instances